src/api/flaskr/service/shifu/outline_funcs.py

```
@@ -249,18 +249,97 @@ def delete_chapter(app, user_id: str, chapter_id: str):
+        # 如果提供了 move_to_parent_id，说明是跨章移动
+        if move_to_parent_id:
+            # 获取目标章节
+            target_chapter = find_node_by_id(outlines, move_to_parent_id)
+            if not target_chapter:
+                raise_error("SHIFU.CHAPTER_NOT_FOUND")
+
+            # 获取目标章节下最大的 lesson_index
+            max_index = 0
+            for child in target_chapter.children:
+                if child.outline.lesson_index > max_index:
+                    max_index = child.outline.lesson_index
```

Pylint (3.3.7)

[refactor] Consider using 'max_index = max(max_index, child.outline.lesson_index)' instead of unnecessary if block

(R1731)